Sources/ComposableArchitecture/Store.swift

```diff
@@ -466,8 +508,7 @@ public final class Store<State, Action> {
 
         \(message)
 
-          Store created on: \(self.initialThread)
-          Action sent on: \(Thread.current)
+          Action sent on: \(Thread.current) instead of \(Thread.main)
```

Comment:
For Stores not using the main queue, this message is misleading. For that case, since it's not possible to obtain the current queue (nor do we have the queue on which it was created), the message could be changed to something like:

```
Action sent on a different DispatchQueue to the one the Store was created on
```

Or, if we were to hold onto the creation `DispatchQueue` in `Store`, we could use:

```
Store was created on \(queue.name), but the Action was sent on a different queue
```

Reply:
Indeed, `Thread.main` a carry on from #805 where checks were restricted to it. I forgot to update the message. 

I think storing the name of the dispatch queue (or even its description while we're at it) is preferable. I'll update the PR.